### PR TITLE
Ports the RBMK from NSV

### PR DESCRIPTION
**Pull request by XeonMations**
## About The Pull Request

Adds in the RBMK nuclear reactor from NSV.

## Why It's Good For The Game

With the recent atmos bugs going on and general engineering incompetence, why not add another engine for engineering to fuck up and blow a hole in the engineering department with.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image
image-1
image
image-1

</details>

## Changelog
:cl: Xeonmations, Aramix
add: Added the RBMK reactor, ported from NSV.
/:cl:

### CODE REVIEW

File: code/game/objects/effects/decals/cleanable/misc.dm [hunk 1]
```diff
@@ -216,3 +216,4 @@
 	icon = 'icons/effects/blood.dmi'
 	icon_state = "xfloor1"
 	random_icon_states = list("xfloor1", "xfloor2", "xfloor3", "xfloor4", "xfloor5", "xfloor6", "xfloor7")
+
```
Comment (Penwin0) [inline code/game/objects/effects/decals/cleanable/misc.dm]:
```suggestion
```

**Comment by Penwin0** [comment]
DNM until proper mapping gets done

**Comment by francinum** [comment]
Why.

**Comment by Rukofamicom** [comment]
God those are some ugly sprites
I don't know what I want a nuclear reactor to look like, but it isn't this. 

I do, on the other hand, want this reactor.

**Comment by Tsar-Salat** [comment]
> God those are some ugly sprites
> I don't know what I want a nuclear reactor to look like, but it isn't this.

Its my understanding that it intends to look like the real-life RBMK control rod & reactor lid.

image

**Comment by PowerfulBacon** [comment]
This shouldn't be mapped in alongside the SM (If it is roundstart mapped it needs to replace the SM, however I would prefer if we stuck to a single roundstart engine type for maps since we know how the SM works and can fix it if things about it don't work), ideally it would be something like a cargo replacement for the antimatter reactor

**Comment by francinum** [comment]
> > Why would you want this mess.
> 
> Better to have a not bugged engine than one that explodes due to atmospherics being bugged! 

Give players a singularity or a tesla if this is an immediate, priority concern. Layering fresh code and an entirely new engine that is immanipulable as equipment like the gravity gen is not a good idea.

File: code/modules/power/rbmk/rbmk.dm [hunk 2]
```diff
@@ -0,0 +1,958 @@
...
+				start_up() //That was the first fuel rod. Let's heat it up.
+			fuel_rods += W
+			W.forceMove(src)
+			radiation_pulse(src, temperature) //Wear protective equipment when even breathing near a reactor!
+		return TRUE
+	if(!slagged && istype(W, /obj/item/sealant))
+		if(power >= 20)
+			to_chat(user, "<span class='notice'>You cannot repair [src] while it is running at above 20% power.</span>")
+			return FALSE
+		if(vessel_integrity >= 350)
```
Comment (JixS4v) [inline code/modules/power/rbmk/rbmk.dm]:
```suggestion
		if(vessel_integrity >= 0.875*initial(vessel_integrity))
```
Seems a bit weird to base some stuff on the initial vessel integrity and other stuff on absolute values, better to standardize.

File: code/modules/power/rbmk/rbmk.dm [hunk 3]
```diff
@@ -0,0 +1,958 @@
...
+			to_chat(user, "<span class='notice'>[src]'s seals are already in-tact, repairing them further would require a new set of seals.</span>")
+			return FALSE
+		if(vessel_integrity <= 0.5 * initial(vessel_integrity)) //Heavily damaged.
+			to_chat(user, "<span class='notice'>[src]'s reactor vessel is cracked and worn, you need to repair the cracks with a welder before you can repair the seals.</span>")
+			return FALSE
+		if(do_after(user, 5 SECONDS, target=src))
+			if(vessel_integrity >= 350)	//They might've stacked doafters
```
Comment (JixS4v) [inline code/modules/power/rbmk/rbmk.dm]:
```suggestion
			if(vessel_integrity >= 0.875*initial(vessel_integrity))	//They might've stacked doafters
```

**Review by JixS4v — CHANGES_REQUESTED** [review]

**Comment by PowerfulBacon** [comment]
I mean, I'm fine with this being buildable to replace something like the TEG, but the supermatter must be the standard engine for all maps.

**Pull request closed without merging**